crimsobot/bot.py

```diff
@@ -189,17 +190,17 @@ async def on_message(self, message: discord.Message) -> None:
                 ])
             )
 
+        # Get context from message for command invocation and text generation
+        ctx = await self.get_context(message, cls=CrimsoContext)
+
         # process commands
-        await self.process_commands(message)
+        await self.invoke(ctx)
```

Comment:
we were inadvertently calling get_context twice? nice 😂

Reply:
yep!